### PR TITLE
New LSP

### PR DESCRIPTION
**Pull request by salamaashoush**
Changes

- Bring the repo back to life and follow kadena.js setup (pnpm, vitest, eslint, tsup).
- Add PR CI checks.
- Use [pactup](https://github.com/kadena-community/pactup) to automatically install pact 5 developments releases.
- Format and lint all files to following kadena.js standards.
- Add vscode command to restart pact lsp

### CODE REVIEW

File: src/languageclient.ts [hunk 1]
```diff
@@ -1,60 +1,51 @@
...
\ No newline at end of file
+import { getPactInstallationInfo, installPact } from '@pact-toolbox/installer';
+import { ExtensionContext, window, workspace } from 'vscode';
+import { Executable, LanguageClient, LanguageClientOptions, ServerOptions } from 'vscode-languageclient/node';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export async function startLanguageClient(_context: ExtensionContext) {
+  const traceOutputChannel = window.createOutputChannel('Pact Language Server trace');
+  // eslint-disable-next-line prefer-const
+  let { versionInfo, isInstalled } = await getPactInstallationInfo({ nightly: true });
```
Comment (EnoF) [inline src/languageclient.ts]: Not sure if we should default to nightly builds. I think most devs would prefer the latest stable version

Comment (salamaashoush) [reply src/languageclient.ts]: The new LSP is not available in the stable version pact4x and the plan is to use development builds of pact5 until it is released and then we can switch

File: src/languageclient.ts [hunk 2]
```diff
@@ -1,60 +1,51 @@
...
+    setTimeout(() => {
+      traceOutputChannel.hide();
+    }, 5000);
+  }
+
+  const run: Executable = {
+    command: versionInfo.pactExecutablePath,
+    args: ['--lsp'],
+    options: {
+      env: process.env,
```
Comment (trendzetter) [inline src/languageclient.ts:28]: is it needed to pass all of the env?

**Comment by trendzetter** [comment]
Do you mind splitting this in separate PR's by concern? formatting, update dependencies, ..

**Review by EnoF — APPROVED** [review]

**Pull request merged**